src/amber/controller/helpers/asset.cr

```diff
@@ -0,0 +1,44 @@
+module Amber::Controller::Helpers
+  module Asset
+    ASSET_MANIFEST = {} of String => String
+
+    {{ run "../../run_macros/generate_asset_helpers" }}
+
+    macro asset_path(path)
+      {% if path.is_a?(StringLiteral) %}
+        {% if Amber::Controller::Helpers::Asset::ASSET_MANIFEST[path] %}
+          {{ "/dist" + Amber::Controller::Helpers::Asset::ASSET_MANIFEST[path] }}
+        {% else %}
+          {{ run "../../run_macros/missing_asset", path }}
+        {% end %}
+      {% elsif path.is_a?(StringInterpolation) %}
+        {% raise <<-ERROR
+        \n
+        The 'asset' macro doesn't work with string interpolation
+        Try this...
+          ▸ Use the 'dynamic_asset' method instead
+        ERROR %}
+      {% else %}
+        {% raise <<-ERROR
+        \n
+        The 'asset' macro requires a literal string like "my-logo.png", instead got: #{path}
+
+        Try this...
+
+          ▸ If you're using a variable, switch to a literal string
+          ▸ If you can't use a literal string, use the 'dynamic_asset' method instead
+
+        ERROR %}
+      {% end %}
+    end
+
+    def dynamic_asset_path(path)
+      fingerprinted_path = Amber::Controller::Helpers::Asset::ASSET_MANIFEST[path]?
```

Comment:
That's a long constant! My first instinct would be to try and trim it down to just `MANIFEST`. Is the `Amber::Controller::Helpers::Asset::` explicit prefix necessary since it is in the module already? 

I think it might also help to make a getter for this, so if people want to inspect the manifest they don't have to go so deep.

Reply:
Do you have a suggestion for this?  I'm not exactly sure what you are intending? :)

Reply:
If line 36 and others can be written like this instead, I think it would clean up the code a lot.

```crystal
      fingerprinted_path = ASSET_MANIFEST[path]? 
```

Additionally, a method to get the manifest without using a constant would be nice for people wanting to do some reflection on it:

```
def manifest
  ASSET_MANIFEST
end
```